modules/web/src/app/core/services/auth/service.ts

```diff
@@ -130,10 +130,16 @@ export class Auth {
     const config = this._appConfigService.getConfig();
     if (config.oidc_logout_url) {
       const logoutUrl = new URL(config.oidc_logout_url);
-      if (logoutUrl.searchParams.has('redirectUri')) {
-        logoutUrl.searchParams.set('redirectUri', this._redirectUri);
-      } else {
-        logoutUrl.searchParams.set('redirect_uri', this._redirectUri);
+      const urlArr = config.oidc_logout_url.split('/');
+      if( urlArr.includes('realms')) {
+          logoutUrl.searchParams.set('post_logout_redirect_uri', this._redirectUri);
+          logoutUrl.searchParams.set('id_token_hint', this.getBearerToken());
+      }else {
+        if (logoutUrl.searchParams.has('redirectUri')) {
+          logoutUrl.searchParams.set('redirectUri', this._redirectUri);
+        } else {
+          logoutUrl.searchParams.set('redirect_uri', this._redirectUri);
+        }
       }
```

Comment:
Realm name is always there since the keycloak clients are always scoped to a realm. The only case otherwise is when you are using some proxy in front of keycloak to obscure the realm name or the keyloak URL. 

So i'm not sure how we are making decision regarding parameters based on the presence of `realm` within the URL?

Reply:
how about we use a new value in the `config` like `config.oidc_provider` ? or something ?
@Waseem826 @ahmedwaleedmalik